sympy/solvers/solveset.py

```diff
@@ -637,6 +638,18 @@ def _solveset(f, symbol, domain, _check=False):
                       if isinstance(s, RootOf)
                       or domain_check(fx, symbol, s)])
 
+            if isinstance(result, FiniteSet) and \
+            not all(isinstance(x, (erf, erfinv, erfc, erfcinv)) for x in result.args):
+                for r in result:
+                    if not checksol(f, symbol, r):
+                        result = FiniteSet(*[x for x in result if x != r])
+                if result is EmptySet():
```

Comment:
Since you are returning the result regardless of whether it is empty or not, a simple `return result` would suffice.